### PR TITLE
En_Fishing (3 non-matching)

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

3 non-matching, 2 due to stack, 1 due to a small register flip. 

I tried to mark all the differences I saw between MM and OoT versions. The Audio_QueueSeqCmd ones are probably unnecessary, and I ignored ones where regs were removed from branches (removing the entire branch when the reg was the only thing). Besides small changes to scoping, the small things added and all the symbol names updated, this is a straight copy from OoT.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.h [hunk 1]
```diff
@@ -9,7 +9,56 @@ typedef void (*EnFishingActionFunc)(struct EnFishing*, GlobalContext*);
 
 typedef struct EnFishing {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x404];
+    /* 0x0144 */ char unk_144[0x004];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.h:12]:
```suggestion
    /* 0x0144 */ UNK_TYPE1 [0x4];
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 2]
```diff
@@ -1,21 +1,204 @@
...
+    /* 0x08 */ FS_EFF_RAIN_SPLASH
+} FishingEffectType;
+
+typedef struct {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3f vel;
+    /* 0x18 */ Vec3f accel;
+    /* 0x24 */ u8 type;
+    /* 0x25 */ u8 timer;
+    /* 0x26 */ char unk_26[0x04];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
    /* 0x26 */ UNK_TYPE unk_26[0x4];
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 3]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+        D_8090CD04 = 20;
+        globalCtx->specialEffects = sFishingEffects;
+        REG(15) = 1; // gTimeIncrement in OoT
+        D_809171FC = 0;
+        D_809171F6 = 10;
+
+        Audio_QueueSeqCmd(0x100100FF);
+
+        if (sLinkAge == 1) {
+            if ((gSaveContext.roomInf[127][2] & 0x7F) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
            if (gSaveContext.roomInf[127][2] & 0x7F) {
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 4]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+                D_809171CC = gSaveContext.roomInf[127][2] & 0x7F;
+            } else {
+                D_809171CC = 40.0f;
+            }
+        } else {
+            if ((gSaveContext.roomInf[127][2] & 0x7F000000) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
            if (gSaveContext.roomInf[127][2] & 0x7F000000) {
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 5]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+};
+
+static Vec3f sRodTipOffset = { 0.0f, 0.0f, 0.0f };
+
+void EnFishing_DrawRod(GlobalContext* globalCtx) {
+    s16 i;
+    f32 spC8;
+    f32 spC4;
+    f32 spC0;
+    Input* input = &globalCtx->state.input[0];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
    Input* input = CONTROLLER1(globalCtx);
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 6]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+    f32 spD0;
+    f32 phi_f16;
+    f32 spC8;
+    s16 i;
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    Vec3f spA8;
+    Vec3f sp9C;
+    Vec3f sp90;
+    Input* input = &globalCtx->state.input[0];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
    Input* input = CONTROLLER1(globalCtx);
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 7]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+            sfxId = NA_SE_EV_OUT_OF_WATER;
+        }
+    }
+
+    Audio_PlayActorSound2(&this->actor, sfxId);
+}
+
+void EnFishing_HandleAquariumDialog(EnFishing* this, GlobalContext* globalCtx) {
+    if (sLinkAge == 1) {
+        if ((gSaveContext.roomInf[127][2] & 0x7F) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
        if (gSaveContext.roomInf[127][2] & 0x7F) {
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 8]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+            if (gSaveContext.roomInf[127][2] & 0x80) {
+                this->actor.textId = 0x40B1;
+            } else {
+                this->actor.textId = 0x4089;
+            }
+        } else {
+            this->actor.textId = 0x40AE;
+        }
+    } else {
+        if ((gSaveContext.roomInf[127][2] & 0x7F000000) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
        if (gSaveContext.roomInf[127][2] & 0x7F000000) {
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 9]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+    s16 phi_v0;
+    s16 spF6;
+    s16 spF4;
+    s16 spF2;
+    s16 spF0;
+    s16 spEE;
+    EnFishing* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+    Player* player = GET_PLAYER(globalCtx);
+    Input* input = &globalCtx->state.input[0];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
    Input* input = CONTROLLER1(globalCtx);
```

File: src/overlays/actors/ovl_En_Fishing/z_en_fishing.c [hunk 10]
```diff
@@ -24,181 +207,5508 @@ const ActorInit En_Fishing_InitVars = {
...
+    Vec3f sp108;
+    Vec3f spFC;
+    s16 headRotTarget;
+    s16 playerShadowAlpha;
+    f32 target;
+    f32 camAtFraction;
+    f32 lureDistXZ;
+    s32 pad;
+    Player* player = GET_PLAYER(globalCtx);
+    Input* input = &globalCtx->state.input[0];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Fishing/z_en_fishing.c]:
```suggestion
    Input* input = CONTROLLER1(globalCtx);
```

**Review by hensldm — COMMENTED** [review]
Some cosmetics

**Comment by AngheloAlf** [comment]
Jenkins is mad because there are misformatted files

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**